### PR TITLE
Allow running container as host user to solve permission problems

### LINKED ISSUE #3: When running on Linux, can't read `application_default_credentials.json` to access gcloud resources
`[Errno 13] Permission denied: '/home/airflow/.config/gcloud/application_default_credentials.json'`

## Cause
The file is present but the UID of the logged-in user (typically `1000`) differs from the UID of the `airflow` user (`999`) in the container. Airflow will not have permission to read the file to access GCP resources with it.

It's the typical Docker-Linux UID/GID mismatch problem when bind mounting directories. (Mac, Windows use network file transfers in the background instead since Docker there runs off a Linux VM.)

## Workaround
Run `chmod +r ~/.config/gcloud/application_default_credentials.json` on your machine.

However other users on the machine now can read this file. This is potentially insecure.

## Proposed Fix
~Include `gcloud_config_path` in `fix_file_permissions()`~

* Copy `~/.config/gcloud` into the container and `chmod +r` instead of volume mounting
* Or set UID/GID to the currently logged-in user when starting the container. This also fixes permissions for the `data`/`dags`/`plugins` directories where the `airflow` user can't modify files written by the user and the user can't modify or read files written by `airflow`.

### PR DESCRIPTION
**Pull request by functicons**
fix #3

### CODE REVIEW

File: README.md
Comment (piotrwrotniak) [inline README.md:211]: nit: Known issues

File: README.md
Comment (piotrwrotniak) [inline README.md:214]:
```suggestion
 running the container with the default user `airflow (999)` and the host directory `~/.config/gcloud/` is missing
```

File: README.md
Comment (piotrwrotniak) [inline README.md:217]:
```suggestion
  because you are running the container with the default user `airflow (999)` and the host file
```

File: README.md
Comment (piotrwrotniak) [inline README.md:215]:
```suggestion
  the `execute` permission for the user.
```

File: README.md
Comment (piotrwrotniak) [inline README.md:218]:
```suggestion
  `~/.config/gcloud/application_default_credentials.json` is missing the `read` permission for the user.
```

**Review by piotrwrotniak — CHANGES_REQUESTED** [review]
Thanks for contributing, left some nits - feel free to disregard.

Please make the PR target `development` branch and wait for current owner approval, we will resync this to main along with the changes we have made internally and are yet unreleased. Thanks!

File: composer_local_dev/environment.py
Comment (ahidalgob) [inline composer_local_dev/environment.py:94]: Won't this break on Windows?

**Review by ahidalgob — CHANGES_REQUESTED** [review]

File: composer_local_dev/environment.py
Comment (functicons) [reply composer_local_dev/environment.py:94]: Good catch. I just confirmed `os.getuid()` is only available on Unix/Linux. https://docs.python.org/3/library/os.html#os.getuid

Any idea how to handle Windows? Should I adde a check for Windows and disallow the feature?

Comment (piotrwrotniak) [reply composer_local_dev/environment.py]: Yeah, that would be great :)

**Review by ahidalgob — APPROVED** [review]

**Pull request merged**
